test/test_bc.py

```diff
@@ -0,0 +1,181 @@
+            initializer = Lump(center=orig, velocity=vel, rhoamp=0.0)
+            wall = AdiabaticSlipBoundary()
+
+            uniform_state = initializer(0, nodes)
+            from functools import partial
+            bnd_norm = partial(discr.norm, p=np.inf, dd=BTAG_ALL)
+
+            bnd_pair = wall.boundary_pair(discr, uniform_state, t=0.0,
+                                          btag=BTAG_ALL, eos=eos)
+            bnd_cv_int = split_conserved(dim, bnd_pair.int)
+            bnd_cv_ext = split_conserved(dim, bnd_pair.ext)
+
+            # check that mass and energy are preserved
+            mass_resid = bnd_cv_int.mass - bnd_cv_ext.mass
+            mass_err = bnd_norm(mass_resid)
+            assert(mass_err == 0.0)
+            energy_resid = bnd_cv_int.energy - bnd_cv_ext.energy
+            energy_err = bnd_norm(energy_resid)
+            assert(energy_err == 0.0)
+
+            # check that exterior momentum term is mom_interior - 2 * mom_normal
+            mom_norm_comp = np.dot(bnd_cv_int.momentum, nhat)
+            mom_norm = nhat * make_obj_array([mom_norm_comp])
+            expected_mom_ext = bnd_cv_int.momentum - 2.0 * mom_norm
+            mom_resid = bnd_cv_ext.momentum - expected_mom_ext
+            mom_err = bnd_norm(mom_resid)
+
+            assert(mom_err == 0.0)
+
+
+@pytest.mark.parametrize("dim", [1, 2, 3])
+def test_slipwall_flux(actx_factory, dim):
+    """Check for zero boundary flux.
+
+    Check for vanishing flux across the slipwall.
+    """
+    actx = actx_factory()
+
+    nel_1d = 4
+
+    from meshmode.mesh.generation import generate_regular_rect_mesh
+
+    mesh = generate_regular_rect_mesh(
+        a=(-0.5,) * dim, b=(0.5,) * dim, n=(nel_1d,) * dim
+    )
+
+    order = 3
+    discr = EagerDGDiscretization(actx, mesh, order=order)
+    nodes = thaw(actx, discr.nodes())
+    eos = IdealSingleGas()
+    orig = np.zeros(shape=(dim,))
+    normal = thaw(actx, discr.normal(BTAG_ALL))
+    normal_mag = actx.np.sqrt(np.dot(normal, normal))
+    nhat_mult = 1.0 / normal_mag
+    nhat = normal * make_obj_array([nhat_mult])
+    wall = AdiabaticSlipBoundary()
+
+    from functools import partial
+    bnd_norm = partial(discr.norm, p=np.inf, dd=BTAG_ALL)
+
+    logger.info(f"Number of {dim}d elems: {mesh.nelements}")
+
+    # for velocities in each direction
+    for vdir in range(dim):
+        vel = np.zeros(shape=(dim,))
+
+        # for velocity directions +1, and -1
+        for parity in [1.0, -1.0]:
+            vel[vdir] = parity
+            initializer = Lump(center=orig, velocity=vel, rhoamp=0.0)
+            uniform_state = initializer(0, nodes)
+
+            bnd_pair = wall.boundary_pair(discr, uniform_state, t=0.0,
+                                          btag=BTAG_ALL, eos=eos)
+
+            # Check the total velocity component normal
+            # to each surface.  It should be zero.  The
+            # numerical fluxes cannot be zero.
```

Comment:
* Isn't the lump zero near the boundary? If so, why is this test meaningful?
* My sense is that the BC only has a right to be true in a convergence sense, not with an absolute tolerance.

Reply:
There is no lump here.   We make a dim-box and set the velocity to non-zero uniform in each direction in turn then check that the momentum .dot. wall normal is zero on the wall.   Each boundary is tested with non-zero tangent and normal velocity components.